waiter/src/waiter/service_description.clj

```diff
@@ -40,6 +40,11 @@
       (str/starts-with? name "MESOS_")
       (re-matches #"^PORT\d*$" name)))
 
+(def environment-variable-schema
+  (s/both (s/constrained s/Str #(<= 1 (count %) 512))
+          #"^[A-Za-z][A-Za-z0-9_]*$"
```

Comment:
Why shouldn't we allow environment variables that start with underscores?

Reply:
Do we have a test for environment variables with hyphens?  This regex seems to disallow them.

Reply:
This is reusing the current regex we have for environment variables. This is part of the current environment variable error messaging: `Keys must be made up of letters, underscores, and numbers and must start with a letter.` So we disallow starting with underscores and use of hyphens.